pkg/R/sparkR.R

```diff
@@ -121,45 +119,38 @@ sparkR.init <- function(
   if (sparkRExistingPort != "") {
     sparkRBackendPort <- sparkRExistingPort
   } else {
+    # TODO: test the random port and retry, or use httpuv:::startServer
+    callbackPort = sample(40000, 1) + 10000
```

Comment:
Can we have a simple loop trying a few increments ? (Something similar to the connection retry we used to have). It would also be good to `stop` with an error message if the `socketConnection` doesn't come up successfully after the retries.